File: java/implementsOfinterface/implementsOfinterface/src/implementsOfinterface/implementsOfinterface/src/DerivedClass.java
// 派生クラス
class DerivedClass extends BaseClass implements SetInterface, GetInterface {	// DerivedClassクラスの定義

	// フィールドの定義
	String str;	// Stringクラスフィールドstr
	int i;		// int型フィールドi

	// メソッドの定義
	// iのセット
	public void set(int i) {	// set(int i)はSetInterfaceから.
		this.i = i;	// this.iにiをセット.
	}

	// strのセット
	public void set(String str) {	// set(String str)はSetInterfaceから.
		this.str = str;	// this.strにstrをセット.
	}

	// iとstrのセット
	public void set(int i, String str) {	// set(int i, String str)はSetInterfaceから.
		this.i = i;	// this.iにiをセット.
		this.str = str;	// this.strにstrをセット.
	}

	// iのゲット
	public int getInt() {	// getInt()はGetInterfaceから.
		return i;
	}

	// strのゲット
	public String getString() {	// getStringはGetInterfaceから.
		return str;
	}

	// 出力
	public void output() {	// outputはBaseClassから.
		System.out.println("i = " + i + ", str = " + str);	// iとstrを出力.
	}

}

File: cs/abstract/abstract/src/abstract/abstract/DerivedClass.cs
// 整数値と文字列を持つ派生クラス
class DerivedClass : AbstractClass // AbstractClassから派生したDerivedClassの定義
{
    // フィールドの定義
    private int i;  // int型フィールドi
    private string str; // string型フィールドstr
    
    // メソッドの定義
    public DerivedClass() : base()  // コンストラクタ
    {
        // 初期化.
        this.i = 0; // iに0をセット.
        this.str = "-"; // strに"-"をセット.
    }

    public DerivedClass(int i, string str) : base(i, str) // コンストラクタ(引数あり)
    {
        // 引数をセット.
        this.i = i; // this.iにiをセット.
        this.str = str; // this.strにstrをセット.
    }
    
    override public void Output()    // 出力.
    {
        // iとstrの出力.
        System.Console.WriteLine("i = " + i + ", str = " + str);    //iとstrの値を出力.
    }

}